flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -1507,12 +1525,13 @@ void OmpAttributeVisitor::ResolveOmpName(
         AddToContextObjectWithDSA(*resolvedSymbol, ompFlag);
       }
     }
-  } else if (ompFlag == Symbol::Flag::OmpCriticalLock) {
-    // Create a new symbol.
+  } else if (ompFlagsRequireNewSymbol.test(ompFlag)) {
```

Comment:
Also, is it required to do this for all flags that require a new symbol or just for Critical?

Reply:
I would prefer restricting to CriticalLock. Not sure whether ObjectEntityDetails are needed for others. So better to restrict UnknownDetails to only CriticalLock.